grafana/Dockerfile

```diff
@@ -1,13 +1,64 @@
+ARG GRAFANA_VERSION=7.5.16
+ARG GOLANG_VERSION=1.19.0-alpine3.16
+
+
+# Fetch the sources
+FROM alpine:3.16 as fetcher
 ARG GRAFANA_VERSION
 
-FROM grafana/grafana:${GRAFANA_VERSION} as grafana-builder
 
-FROM alpine:3.16.2 as grafana
+WORKDIR /build
+ADD https://github.com/grafana/grafana/archive/refs/tags/v${GRAFANA_VERSION}.tar.gz /grafana.tar.bz2
+RUN tar -xf /grafana.tar.bz2 -C /build --strip-components 1
+
+# Build UI components
+FROM node:14.15.5-alpine3.13 as js-builder
+WORKDIR /usr/src/app/
+COPY --from=fetcher /build/package.json /build/yarn.lock ./
+COPY --from=fetcher /build/packages packages
+
+RUN apk add --no-cache git
+
+RUN yarn install --pure-lockfile --no-progress
+
+COPY --from=fetcher /build/tsconfig.json /build/.eslintrc /build/.editorconfig /build/.browserslistrc /build/.prettierrc.js ./
+COPY --from=fetcher  /build/public public
+COPY --from=fetcher  /build/tools tools
+COPY --from=fetcher  /build/scripts scripts
+COPY --from=fetcher  /build/emails emails
+
+ENV NODE_ENV production
+RUN yarn build
+
+# Build go components
+FROM golang:${GOLANG_VERSION} as go-builder
+
+RUN apk add --no-cache gcc g++
+
+WORKDIR $GOPATH/src/github.com/grafana/grafana
+
+COPY --from=fetcher /build/go.mod ./
+COPY --from=fetcher /build/go.sum ./
+
+COPY --from=fetcher /build/pkg pkg
+
+RUN go mod verify
+
+# COPY --from=fetcher /build/pkg pkg
+COPY --from=fetcher /build/build.go /build/package.json ./
```

Comment:
is the package.json needed?

Reply:
Yeah its used in the main to read the version: https://github.com/grafana/grafana/blob/c0e2ad126c0e83928f3a358e159f442f21cf8d08/build.go#L172